services/shhext/dedup/deduplicator.go

```diff
@@ -35,18 +44,26 @@ func NewDeduplicator(keyPairProvider keyPairProvider, db *leveldb.DB) *Deduplica
 // Deduplicate receives a list of whisper messages and
 // returns the list of the messages that weren't filtered previously for the
 // specified filter.
-func (d *Deduplicator) Deduplicate(messages []*whisper.Message) []DeduplicateMessage {
+func (d *Deduplicator) Deduplicate(messages []*protocol.StatusMessage) []DeduplicateMessage {
+
 	result := make([]DeduplicateMessage, 0)
 	selectedKeyPairID := d.keyPairProvider.SelectedKeyPairID()
 
 	for _, message := range messages {
-		if has, err := d.cache.Has(selectedKeyPairID, message); !has {
+		whisperMessage := message.TransportMessage
```

Comment:
...do this transformation by the caller. Then we could have the signature like this:
```go
func (d *Deduplicator) Deduplicate(messages []*DeduplicateMessage)
```

and just do this part:
```go
// redacted
for _, message := range messages {
    // redacted
    message.Metadata.DedupID = d.cache.KeyToday(selectedKeyPairID, message.Message)
}
```

Reply:
done!